tests/jballerina-unit-test/src/test/resources/test-src/statements/variabledef/record-variable-definition-stmt.bal

```diff
@@ -743,6 +743,20 @@ function testReadOnlyRecordWithMappingBindingPatternInVarDecl() {
     assertEquality(<ReadOnlyRecord> {x: [1, 2], y: "s1"}, b1);
 }
 
+function recordVariableWithAnonymousType() returns [string, boolean]|error {
+    record {|
+        string name;
+        boolean married;
+    |} {name, married} = check getPersonRecord().ensureType();
```

Comment:
Do we have tests for tuples and errors?

Reply:
We currently do not allow a record literal of that type with the binding pattern[example 1], so we use the 'typedesc' of the binding pattern when used with the 'ensure' type.

```ballerina
function recordVariableWithAnonymousType() returns [string, boolean] {
    record {|
        string name;
        boolean married;
    |} {name, married} = {name: "Jack", married: true};
    return [name, married];
}
```

Reply:
Are there any other scenarios where we might need to use the 'typedesc' of a binding pattern?

Reply:
I don't get the relevance? I meant tests for list and error binding patterns with tuples (and maybe arrays) and errors that have anonymous records/mapping binding patterns as members?

Reply:
Could you please provide an example for this?

Reply:
```ballerina
    [int, record {|
        string name;
        boolean married;
    |}] [id, {name, married}] = check a.ensureType();
```

Reply:
Why do we need to generate a typedesc for?
record {|
        string name;
        boolean married;
    |} in LHS

Reply:
The typedesc used as the argument for `ensureType` is 

```
[int, record {|
        string name;
        boolean married;
    |}]
```

which has the record as the member.

Why shouldn't we generate anyway, when there is a record typedesc in the code?

Reply:
Yes, in this scenario we generate the 'typedesc' and create the tuple variable. However, we encounter this problem when visiting the record variable.
We have these types of tests in https://github.com/ballerina-platform/ballerina-lang/blob/master/tests/jballerina-unit-test/src/test/resources/test-src/statements/variabledef/tuple-variable-definition.bal

Reply:
If the type node wasn't visited for "tuple variables", we'd have a similar issue, right? Thus, my concern about having a test for tuples/errors too.

Reply:
In the current implementation we visited the typenode of tuple variables https://github.com/ballerina-platform/ballerina-lang/blob/3eda82f07b8e47b8ade714deae35dc2574504f1e/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java#L1522

Reply:
Yes, my point is, if we have tests for them?

Reply:
Yes. We have tests in  https://github.com/ballerina-platform/ballerina-lang/blob/master/tests/jballerina-unit-test/src/test/resources/test-src/statements/variabledef/tuple-variable-definition.bal

Reply:
Here, typeDesc of record type is used to create TupleValueImp. So do we need to add a test with ensureType?

Reply:
add test for this.